samples/excel/90-scenarios/performance-optimization.yaml

```
@@ -11,75 +11,132 @@ script:
-                        cell.values = [[i * j * Math.random()]];
-                        if (untrack) {
-                            cell.untrack();
-                        }
-                    }
+            const startTime = Date.now();
+            console.log("Starting...");
+
+            // If other parts of the sample have toggled screen printing off, this will stop screen updating until context.sync is called.
+            if (pauseScreenPainting) {
+              context.application.suspendScreenUpdatingUntilNextSync();
+            }
+
+            for (let i = 1; i < ROW_COUNT; i++) {
+              for (let j = 1; j < COLUMN_COUNT; j++) {
+                let cell = sheet.getCell(i, j);
+                cell.values = [[i * j * Math.random()]];
+
+                // If other parts of the sample have toggled tracking off, we will avoid tracking this range and having to manage the proxy objects.
+                // For more information, see https://docs.microsoft.com/office/dev/add-ins/concepts/resource-limits-and-performance-optimization#untrack-unneeded-proxy-objects
+                if (untrack) {
+                  cell.untrack();
                 }
+              }
+            }
 
-                await context.sync();
+            await context.sync();
 
-                console.log(`Ending. Adding ${rowCount * columnCount} cells took ${Date.now() - startTime} milliseconds`);
-            });
+            console.log(`Ending. Adding ${ROW_COUNT * COLUMN_COUNT} cells took ${Date.now() - startTime} milliseconds`);
+          });
         }
 
         function toggleScreenPainting() {
-            pauseScreenPainting = !pauseScreenPainting;
-            console.log(`pauseScreenPainting is ${pauseScreenPainting}`);
+          pauseScreenPainting = !pauseScreenPainting;
+          if (pauseScreenPainting) {
+            console.log("Screen updating will be paused when you press Refresh Data.");
+          } else {
+            console.log("Screen updating will happen as normal.");
+          }
         }
 
         function toggleTracking() {
-            untrack = !untrack;
-            console.log(`untrack is ${untrack}`);
+          untrack = !untrack;
+          if (untrack) {
+            console.log("Proxy Range objects will not be tracked when you press Refresh Data.");
+          } else {
+            console.log("Proxy Range objects will  be tracked as normal.");
+          }
+        }
+
+        async function toggleCalculationMode() {
+          await Excel.run(async (context) => {
+            context.application.load("calculationMode");
+            await context.sync();
+
+            let calculationMode = context.application.calculationMode;
+            if (calculationMode === Excel.CalculationMode.automatic) {
+              context.application.calculationMode = Excel.CalculationMode.manual;
+              console.log(`Calculation Mode is now ${Excel.CalculationMode.manual}`);
```

For consistency with HTML tab
```suggestion
              console.log(`Calculation mode is now ${Excel.CalculationMode.manual}`);
```